src/su.c

```diff
@@ -743,7 +743,7 @@ static void save_caller_context (char **argv)
 	 * most error messages.
 	 */
 	Prog = Basename (argv[0]);
-	log_set_progname(Prog);
+	log_set_progname("su");
```

Comment:
You should probably `Prog = "su"`.  Otherwise, other uses of `Prog` in fprintf(3) calls will have similar issues.  Is it useful to read argv[0] at all?